hmtk/seismicity/catalogue.py

```diff
@@ -419,3 +415,74 @@ def get_magnitude_time_distribution(self, magnitude_bins, time_bins,
             ysigma=self.data['sigmaMagnitude'],
             normalisation=normalisation,
             number_bootstraps=bootstrap)
+
+        
+    def concatenate(self, catalogue):
+        """
+        This method attaches one catalogue to the current one
+
+        :parameter catalogue: 
+            An instance of :class:`htmk.seismicity.catalogue.Catalogue`
+        """
+
+        atts = getattr(self, 'data')
+        attn = getattr(catalogue, 'data')
+        data = _merge_data(atts, attn)
+
+        if data is not None:
+            setattr(self, 'data', data)
+            for attrib in vars(self):
+                atts = getattr(self, attrib)
+                attn = getattr(catalogue, attrib)
+                if attrib is 'end_year':
+                    setattr(self, attrib, max(atts, attn))
+                elif attrib is 'start_year':
+                    setattr(self, attrib, min(atts, attn))
+                elif attrib is 'data':
+                    pass
+                elif attrib is 'number_earthquakes':
+                    setattr(self, attrib, atts+attn)
+                elif attrib is 'processes':
+                    if atts != attn:
+                        raise ValueError('The catalogues cannot be merged' +
+                                         ' since the they have' +
+                                         ' a different processing history')
+                else:
+                    print attrib
+                    raise ValueError('unknown attibute')
```

Comment:
One concern I have is if merging two catalogue together the events might not follow a chronological order. I think it could be useful by default to run the sort_catalogue_chronologically on the merge catalogue before returning control to the user.